syncany-lib/src/main/java/org/syncany/database/dao/FileVersionSqlDao.java

```diff
@@ -149,6 +149,23 @@ public void removeFileVersions(Map<FileHistoryId, FileVersion> purgeFileVersions
 		}
 	}
 
+	public void removeSpecificFileVersions(Map<FileHistoryId, List<FileVersion>> purgeFileVersions) throws SQLException {
+		if (purgeFileVersions.size() > 0) {
+			try (PreparedStatement preparedStatement = getStatement(connection, "fileversion.delete.all.removeSpecificFileVersionsByIds.sql")) {
+				for (FileHistoryId purgeFileHistoryId : purgeFileVersions.keySet()) {
+					for (FileVersion purgeFileVersion : purgeFileVersions.get(purgeFileHistoryId)) {
```

Comment:
For future usage: [`ListMultimap`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/ListMultimap.html) from guava makes the usage of Maps with multiple entries per key easier.